### ISSUE TITLE
IntEnum non-iterable error

### ISSUE THREAD

**Issue by saifrim** [PyCQA/pylint#2598]
### Steps to reproduce
```
from enum import IntEnum


class Color(IntEnum):
    RED = 1
    GREEN = 2
    BLUE = 3


for c in Color:
    print c
```

### Current behavior
`E: 10, 9: Non-iterable value Color is used in an iterating context (not-an-iterable)`

### Expected behavior
If I do the mixin explicitly, Pylint doesn't complain
```
...
class Color(int, Enum):
...
```

### pylint --version output
pylint 1.9.3,
astroid 1.6.5
Python 2.7.10 (default, Oct  6 2017, 22:29:07)
[GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.31)]

There is an old issue on this: https://github.com/PyCQA/pylint/issues/1539

**Comment by saifrim** [comment]
Found that `__reversed__` is not properly checked also
```
from enum import IntEnum


class Color(IntEnum):
    RED = 1
    GREEN = 2
    BLUE = 3


for c in reversed(Color):
    print c
```
### Current behavior
`E: 10, 9: bad-reversed-sequence / The first reversed() argument is not a sequence`

**Comment by PCManticore** [comment]
Thanks for the report @saifrim !
